molter/help.py

```diff
@@ -0,0 +1,197 @@
+
+    def register(self) -> None:
+        """Register the help command in dis-snek."""
+        if not isinstance(self.cmd.callback, functools.partial):
+            # prevent wrap-nesting
+            self.cmd.callback = functools.partial(self.cmd.callback, self)
+
+        # replace existing help command if found
+        if "help" in self._client.commands:
+            log.warning("Replacing existing help command.")
+            del self._client.commands["help"]
+
+        self._client.add_message_command(self.cmd)  # type: ignore
+
+    async def send_help(self, ctx: dis_snek.MessageContext, cmd_name: str | None) -> None:
+        """
+        Send a help message to the given context.
+
+        args:
+            ctx: The context to use
+            cmd_name: An optional command name to send help for.
+        """
+        await self._callback.callback(ctx, cmd_name)  # type: ignore
+
+    @molter.msg_command(name="help")
+    async def _callback(self, ctx: dis_snek.MessageContext, cmd_name: str = None) -> None:
+        if cmd_name:
+            return await self._help_specific(ctx, cmd_name)
+        await self._help_list(ctx)
+
+    async def _help_list(self, ctx: dis_snek.MessageContext) -> None:
+        cmds = await self._gather(ctx)
+
+        output = []
+        for cmd in cmds.values():
+            _temp = self._generate_command_string(cmd, ctx)
+            _temp += f"\n{cmd.brief}"
+
+            output.append(self._sanitise_mentions(_temp))
+        if len("\n".join(output)) > 500:
+            paginator = Paginator.create_from_list(self._client, output, page_size=500)
+            paginator.default_title = self.embed_title.format(username=self._client.user.username)
+            await paginator.send(ctx)
+        else:
+            embed = Embed(
+                title=self.embed_title.format(username=self._client.user.username),
+                description="\n".join(output),
+                color=dis_snek.BrandColors.BLURPLE,
+            )
+            await ctx.reply(embeds=embed)
+
+    async def _help_specific(self, ctx: dis_snek.MessageContext, cmd_name: str) -> None:
+        cmds = await self._gather(ctx)
+
+        if cmd := cmds.get(cmd_name.lower()):
+            _temp = self._generate_command_string(cmd, ctx)
+            _temp += f"\n{cmd.help}"
+            await ctx.reply(self._sanitise_mentions(_temp))
+        else:
+            await ctx.reply(self.not_found_message.format(cmd_name=cmd_name))
+
+    async def _gather(self, ctx: dis_snek.MessageContext | None = None) -> dict[str, molter.MolterCommand]:
+        """
+        Gather commands based on the rules set out in the class attributes.
+
+        args:
+            ctx: The context to use to establish usability.
+
+        returns:
+            dict[str, MolterCommand]: A list of commands fit the class attribute configuration.
+        """
+        out: dict[str, molter.MolterCommand] = {}
+
+        for cmd in self._client.commands.values():
```

Comment:
It shouldn't be assumed that every command here is a Molter command... or at least it may be a good idea just to add a filter for normal message commands *just* in case.

Reply:
My understanding was MolterCommands were replacing snek's message commands, is this not the case?

Reply:
Added a filter